### PR TITLE
Prettified the code

### PR DESCRIPTION
**Pull request by osher-sade**

### CODE REVIEW

File: src/data-version/data-version-middleware.ts [hunk 1]
```diff
@@ -1,29 +1,43 @@
-import {PolarisBaseContext} from "@enigmatis/polaris-common"
+import { PolarisBaseContext } from '@enigmatis/polaris-common';
 
-const dataVersionHeaderName: string = "data-version";
+const dataVersionHeaderName: string = 'data-version';
```
Comment (chenshoo) [inline src/data-version/data-version-middleware.ts]: create header names in common and import from there

File: src/data-version/data-version-middleware.ts [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+    info: any,
+) => {
+    if (context.logger) {
+        context.logger.debug('Data version middleware started job', { context });
+    }
     const result = await resolve(root, args, context, info);
     let finalResult;
-    if (!root && context.dataVersion && !isNaN(context.dataVersion)) { // assert that it has no root (so it is the root)
+    if (!root && context.dataVersion && !isNaN(context.dataVersion)) {
+        // assert that it has no root (so it is the root)
```
Comment (chenshoo) [inline src/data-version/data-version-middleware.ts]: remove doc

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
-import {softDeletedMiddleware} from "./soft-delete/soft-delete-middleware"
-import {realitiesMiddleware} from "./realities/realities-middleware";
-import {ExtensionsListener, ExtensionsPlugin} from './plugins/extensions-plugin';
+import {
+    dataVersionMiddleware,
+    initContextForDataVersion,
+} from './data-version/data-version-middleware';
+import { softDeletedMiddleware } from './soft-delete/soft-delete-middleware';
+import { realitiesMiddleware } from './realities/realities-middleware';
+import { ExtensionsPlugin } from './plugins/extensions-plugin';
```
Comment (chenshoo) [inline src/index.ts]: change to export directly

File: test/data-version/data-version-middleware.test.ts [hunk 4]
```diff
@@ -1,85 +1,72 @@
...
-                undefined, {}, context, {});
-            expect(result).to.deep.equal(objects);
+            const result = await dataVersionMiddleware(resolve, undefined, {}, context, {});
+            expect(result).toEqual(objects);
         });
-        it ('a single entity is resolved, no filter should be applied', async()=>{
-            const context: PolarisBaseContext = {dataVersion: 3};
-            const objects = {title: 'moshe', dataVersion: 2};
-            const resolve = async (root: any, args: any, context: any, info: any) => {
+        it('a single entity is resolved, no filter should be applied', async () => {
```
Comment (chenshoo) [inline test/data-version/data-version-middleware.test.ts]: filter should be applied if its root

File: src/data-version/data-version-middleware.ts [hunk 5]
```diff
@@ -1,29 +1,43 @@
...
         if (result instanceof Array) {
-            finalResult = result.filter(entity => entity.dataVersion && context.dataVersion ?
-                entity.dataVersion > context.dataVersion : entity);
+            finalResult = result.filter(entity =>
+                entity.dataVersion && context.dataVersion
+                    ? entity.dataVersion > context.dataVersion
+                    : entity,
+            );
         } else {
```
Comment (chenshoo) [inline src/data-version/data-version-middleware.ts]: add check on single entity from root

**Review by chenshoo — CHANGES_REQUESTED** [review]

File: src/listeners/extensions-listener.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                response.extensions.irrelevantEntities = returnedExtensions.irrelevantEntities;
+            }
+        }
+
+        if (returnedExtensions.globalDataVersion) {
+            response.extensions.dataVersion = returnedExtensions.globalDataVersion;
+        } else {
+            if (this.dataVersionRepository) {
+                try {
+                    const result = await this.dataVersionRepository.find();
```
Comment (chenshoo) [inline src/listeners/extensions-listener.ts]: use findOne
if (result)
result.value

File: src/realities/realities-middleware.ts [hunk 7]
```diff
@@ -1,22 +1,31 @@
...
+    context: PolarisGraphQLContext,
+    info: any,
+) => {
     const result = await resolve(root, args, context, info);
     const operationalRealityId: number = 0;
-    context.realityId = context.realityId ? context.realityId : 0;
-    const noRealityIdOrSameAsHeader = (entity: any) => entity.realityId == undefined || entity.realityId == context.realityId;
-    if (!root) { // assert that its a root resolver
+    context.requestHeaders.realityId = context.requestHeaders.realityId
+        ? context.requestHeaders.realityId
```
Comment (chenshoo) [inline src/realities/realities-middleware.ts]: context.requestHeaders.realityId || 0

File: src/realities/realities-middleware.ts [hunk 8]
```diff
@@ -1,22 +1,31 @@
...
+        : 0;
+    const noRealityIdOrSameAsHeader = (entity: any) =>
+        entity.realityId === undefined || entity.realityId === context.requestHeaders.realityId;
+    if (!root) {
         if (result instanceof Array) {
             return result.filter(noRealityIdOrSameAsHeader);
         } else {
-            if (noRealityIdOrSameAsHeader(result))
-                return result;
+            if (noRealityIdOrSameAsHeader(result)) return result;
```
Comment (chenshoo) [inline src/realities/realities-middleware.ts]: curly braces, run lint:fix

**Review by chenshoo — CHANGES_REQUESTED** [review]

File: src/listeners/extensions-listener.ts (hunk 6 above)
Comment (osher-sade) [reply src/listeners/extensions-listener.ts]: should i pass argument to `findOne`?

**Review by chenshoo — APPROVED** [review]

**Pull request merged**